coordinator/src/main/scala/filodb.coordinator/queryplanner/MultiPartitionPlanner.scala

```diff
@@ -766,16 +766,16 @@ class MultiPartitionPlanner(val partitionLocationProvider: PartitionLocationProv
     val rhsQueryContext = qContext.copy(origQueryParams = qContext.origQueryParams.asInstanceOf[PromQlQueryParams].
       copy(promQl = LogicalPlanParser.convertToQuery(logicalPlan.rhs)))
 
-    val lhsExec = this.materialize(logicalPlan.lhs, lhsQueryContext)
-    val rhsExec = this.materialize(logicalPlan.rhs, rhsQueryContext)
+    val lhsPlanRes = walkLogicalPlanTree(logicalPlan.lhs, lhsQueryContext)
+    val rhsPlanRes = walkLogicalPlanTree(logicalPlan.rhs, rhsQueryContext)
```

Comment:
I see, the materialize by default returns just the head. Awesome find 👍 

`walkLogicalPlanTree(logicalPlan, qContext).plans.head`